### PR TITLE
Patch 2

### PR DESCRIPTION
**Pull request by RobinNagpal**

### CODE REVIEW

File: generated/summaries/data_types.md [hunk 1]
```diff
@@ -5,16 +5,155 @@ This is the course header. This will be added on top of every page. Go to [DoDAO
  
  ## Blockchain Basics
```
Comment (RobinNagpal) [inline generated/summaries/data_types.md:6]: Change the heading to "Data Types"

File: generated/summaries/data_types.md [hunk 2]
```diff
@@ -5,16 +5,155 @@ This is the course header. This will be added on top of every page. Go to [DoDAO
...
  
+ **Typed language**        
+- Dynamically-Typed Languages
+  * Dynamically-typed languages are the languages where the interpreter assigns variables a data type at runtime based on the variable's value at that time.
+  * Some Examples of Dynamically Typed Languages are:- JavaScript, Python, Perl, Ruby, etc.
+
+- Statically-Typed Languages
+  * Statically-typed languages are the languages where variable types are known at compile time i.e. the type checking is done at compile time.
```
Comment (RobinNagpal) [inline generated/summaries/data_types.md:14]: Would be good to add couple of more points under both dynamic and static languages like static languages are less error prone etc.

File: generated/summaries/data_types.md [hunk 3]
```diff
@@ -5,16 +5,155 @@ This is the course header. This will be added on top of every page. Go to [DoDAO
...
+    * Integers can be declared using keyword let, and explicitly defining the variable size and it's type of signed or unsigned integer.
+    * For example, we can declare a 64-bit signed and 32-bit unsigned integer in the following ways respectively :- 
+    ```
+        1. let mut x: i64 = -20;
+        2. let mut x: u32 = 20;
+    ```
+    * The isize and usize types depend on the architecture of your build, which is denoted in the table as “arch”: 64 bits if you’re on a 64-bit architecture and 32 bits if you’re on a 32-bit architecture.
+    * The number literals that can be multiple numeric types allow a type suffix, such as `20u32`, to designate the type. Number literals can also use `_` as a visual separator to make the number easier to read, such as `1_000` , which will have the same value as if you had specified `1000`.
+    - Integer Overflow
+      * Let’s say you have a variable of type u8 that can hold values between 0 and 255. If you try to change the variable to a value outside of that range, such as 256, integer overflow will occur, which can result in one of two behaviors:-
```
Comment (RobinNagpal) [inline generated/summaries/data_types.md:36]: is `integer overflow will occur` a panic or an Error ? What happens to the program execution?

File: generated/summaries/data_types.md [hunk 4]
```diff
@@ -5,16 +5,155 @@ This is the course header. This will be added on top of every page. Go to [DoDAO
...
+
+- Weak Keywords
+  * These keywords have special meaning only in certain contexts.
+  * For example, it is possible to declare a variable or method with the name `union`.
+ 
+ **Bit Manipulation**        
+- Bit Mnipulation is strictly done by rust compilers for storing data in variables, for two's compliment and many other purposes.
+ 
+ **Strings**        
+- String
```
Comment (RobinNagpal) [inline generated/summaries/data_types.md:147]: We should expand on String and explain it in detail as "Strings" are a very important topic

File: generated/summaries/data_types.md (hunk 3 above)
Comment (parth1614) [reply generated/summaries/data_types.md:36]: - while compiling the program in debug mode, panic occurs at the runtime, and the execution is failed.

- while compiling the program with the '--release' flag, panic occurs, but the execution doesn't stop; instead, 2s complement occurs. 
    For Example:- For example 2's complement of binary number `10010 is (01101) + 1 = 01110`

File: generated/summaries/data_types.md (hunk 4 above)
Comment (parth1614) [reply generated/summaries/data_types.md:147]: Totally Agreed, I'll get to work

File: generated/summaries/data_types.md (hunk 2 above)
Comment (parth1614) [reply generated/summaries/data_types.md:14]: Makes sense, On it..

**Pull request merged**